src/libs/DraftCommentUtils.ts

```diff
@@ -0,0 +1,49 @@
+import type {OnyxEntry} from 'react-native-onyx';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '@src/ONYXKEYS';
+
+const draftCommentMap: Record<string, OnyxEntry<string>> = {};
+Onyx.connect({
+    key: ONYXKEYS.COLLECTION.REPORT_DRAFT_COMMENT,
+    callback: (value, key) => {
+        if (!key) {
+            return;
+        }
+
+        const reportID = key.replace(ONYXKEYS.COLLECTION.REPORT_DRAFT_COMMENT, '');
+        draftCommentMap[reportID] = value;
+    },
+});
```

Comment:
Can we do something like below? If we don't have `waitForCollectionCallback: true`, Onyx will do a loop for each key present in `REPORT_DRAFT_COMMENT`, which over the time can produce performance issues.

```suggestion
Onyx.connect({
    key: ONYXKEYS.COLLECTION.REPORT_DRAFT_COMMENT,
    waitForCollectionCallback: true,
    callback: (draftReportComments) => {
        draftCommentMap = draftReportComments;
    },
});
```

Later you can use `draftCommentMap` like:

```ts
function getDraftComment(reportID: string): OnyxEntry<string> {
    return draftCommentMap[`${ONYXKEYS.COLLECTION.REPORT_DRAFT_COMMENT}${reportID}`];
}
```

Reply:
Great suggestion @hurali97, thanks a lot! Done ✅